rfc/0009-qsl-structure.md

```diff
@@ -0,0 +1,134 @@
+### Top-level Structure
+Keep in mind that the modules listed here are the names of the current Fabric API modules, and the final internal structure of QSL **will not** be the same as Fabric API.
+- `core` - All modules here are implicitly depended on by mods by default, even if they do not use any code from them. Mods can choose to opt-out from this, but if they choose to use the fat QSL dependency they are responsible for ensuring these modules don't get transitively pulled in.
+    - Lifecycle Events
+    	- Here because of the Networking API
+    	- Some parts might be split into `entities` and `world`
+    - Networking API (maybe with relevant parts split into other modules)
+        - Registry Sync (merged in)
+        - (future) Proper Handshake API
+    - Crash Reports
+    - Resource Loader
+- `common` - "Dumping ground" of shared code needed across modules. This is a little more complex, and uses github's code owners to define ownership of individual modules instead of the library as a whole. Modules should only be here if 100% needed, and if enough similar code goes here it can be split to its own library.
+	- None currently
+- `transfer`
+    - API Lookup API
+    - fluid api
+    - item transfer api
+- `config_and_admin` // better name needed
+    - Command API
+    - Game Rule API
+    - (future) Permissions API
+- `rendering`
+	- BlockRenderLayer Registration
+	- Indigo
+	    - need an easy way to exclude this
+	- Models
+	- Renderer API
+		- This may need a better name in a spot?
+	- Rendering
+	- Renderer Registries
+	- Rendering Fluids
+	- Textures
+	- Object Builder API (FabricModelPredicateProviderRegistry)
+- `data` - Tools for working with, generating, or consuming data (e.g. datapacks, advancements, recipes, tags)
+    - (future) Recipe API
+    - (future) Datagen APIs
+    - Rendering Data Attachment
+    - Tag Extensions
+    - Loot Tables
+    - Object Builder API (Criterion)
+- `world` // Needs better name to avoid confusion with Worldgen
+	- `block` // these groups are 100% internal; they are in the same repo and all managed by the `world` team
+		- BlockEntity Networking (merged in)
+    	- Object Builder API (FabricBlockSettings, FabricBlockEntityBuilder and FabricMaterialBuilder)
+    	- (future) Block Extensions (think IForgeBlock)
+    	- Content Registries (Flammable Block Registry)
+	- `item`
+    	- Item API
+    	- Content Registries (Fuel Registry, Composting Registry)
+    	- Tool Attribute API
+    	- (future) Item Extensions (think IForgeItem)
+	- `entity`
+    	    	- Object Builder API (EntityType Builder, Trades, Villager Professions and Types)
+    	    	- Entity Events
+    - `other`
+    	- Particles
+    	- Object Builder API (Point of interest)
+    	- Interaction Events
+        - Could probably use refactoring, but as it stands this absolutely deals with events happening in the world
+    	- Key Binding API
```

Comment:
Key Binding API should not be in the "ingame" library.

After quickly checking the code key bindings can work in screens that are not in-game.

In Keyboard we can see the condition `if (this.client.currentScreen == null || this.client.currentScreen.passEvents)` before changing states of key bindings.